addons/block_code/ui/block_canvas/node_block_canvas/node_block_canvas.gd

```diff
@@ -51,6 +51,7 @@ func _generate_script_from_entry_blocks(entry_statement: String, entry_blocks: A
 
 		if next_block != null:
 			var generator: InstructionTree = InstructionTree.new()
+			InstructionTree.IDHandler.reset()
```

Comment:
Well, any top level variables are user defined so they should be different from each other. But I realize from your other comment that I didn't really check that anywhere.

And you're right, I didn't really want to pass the instruction tree into `get_instruction_node()` but I guess the generated code is dependant on the state of the instruction tree so it's fine. I'll keep track of the counts non-statically. I think we'll change the code generation once we integrate the new block resources anyways.

Reply:
Right, probably good enough as is. However, maybe I'm misunderstanding how entry blocks are handled, but shouldn't this be reset once per entry statement to be scoped correctly? For example, suppose I have multiple `ready_block`s. `generate_script_from_current_window` will collect them into an array keyed by the `func _ready():` so that they can be serialized into a single function body. Within that function, we would want to reset the ID counter once so the names are unique, not once per entry block.

If I'm understanding correctly, you could put 2 `On Ready` blocks on the canvas and have them each load a sound. The sound variables would end up both being named `sound_0` and conflict. Am I reading that correctly?

Reply:
Oh, you're right. For some reason I thought the `_generate_script_from_entry_blocks` was referring to all of the entry blocks, not the ones with the same names. Resetting the counter at the beginning of `_generate_script_from_entry_blocks` before the loop should work properly.